### PR TITLE
feat(MessageBar): Added expand callback to the MessageBar component. (Ready for Review)

### PR DESCRIPTION
**Pull request by AntonVasserman**
## Previous Behavior

Currently we can't use a callback when we expand the message bar.

## New Behavior

I have added a callback to be called whenever we use the expand button.

## Related Issue(s)

This change should help us solve an issue when using MessageBar with links in it.
Here are 2 related issues for FluentUI due to which we are adding the current functionality:
https://github.com/microsoft/fluentui/issues/8776
https://github.com/microsoft/fluentui/issues/30162
(the functionality in this PR should help us work around the issue mentioned in the both above issues)

### CODE REVIEW

**Comment by ling1726** [comment]
@micahgodbolt  This PR is for the v8 MessageBar

**Review by spmonahan — APPROVED** [review]

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/86455f77ea6d207d642d2a501cf7642694ce58ca'>86455f77ea6d207d642d2a501cf7642694ce58ca</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-84e5924e-42c6-44ae-a13f-981d707ed986.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-84e5924e-42c6-44ae-a13f-981d707ed986.json:1]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=30221&commitId=5ea9732deaa4fba875912ad405ca390c5720dc4e&env=prod&clientType=FLUENTUI)


  <details><summary><b>Button Split (compat)</b> 3 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Button Split (compat).Primary Disabled.hover split.chromium.png|1695| Changed|
|Button Split (compat).Primary Disabled.hover main.chromium.png|1695| Changed|
|Button Split (compat).Primary Disabled.default.chromium.png|1695| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Pull request merged**